File: RegexFileSorter/Converters/IntToGridLengthConverter.vb
Imports System.Globalization

Namespace Converters
	Public Class IntToGridLengthConverter
		Implements IValueConverter

		Public Function Convert(value As Object, targetType As Type, parameter As Object, culture As CultureInfo) As Object Implements IValueConverter.Convert
			Return New GridLength(Math.Min(10, Math.Max(1, CInt(value))), GridUnitType.Star)
		End Function

		Public Function ConvertBack(value As Object, targetType As Type, parameter As Object, culture As CultureInfo) As Object Implements IValueConverter.ConvertBack
			Throw New NotImplementedException()
		End Function

	End Class
End Namespace

File: RegexFileSorter/Converters/IntToGridLengthConverter.cs
using System;
using System.Globalization;
using System.Windows;
using System.Windows.Data;

namespace RegexFileSorter.Converters
{
    internal class IntToGridLengthConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, CultureInfo culture) => new GridLength(Math.Min(10, Math.Max(1, (int)value)), GridUnitType.Star);

        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture) => throw new NotImplementedException();
    }
}